src/opm/input/eclipse/Schedule/BCProp.cpp

```diff
@@ -0,0 +1,173 @@
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.
+*/
+
+#include <string>
+
+#include <opm/input/eclipse/Deck/Deck.hpp>
+#include <opm/input/eclipse/Parser/ParserKeywords/B.hpp>
+#include <opm/input/eclipse/Schedule/BCProp.hpp>
+
+namespace Opm {
+namespace {
+
+namespace fromstring {
+
+BCType bctype(const std::string& s) {
+    if (s == "RATE")
+        return BCType::RATE;
+
+    if (s == "FREE")
+        return BCType::FREE;
+
+    if (s == "DIRICHLET")
+        return BCType::DIRICHLET;
+
+    if (s == "THERMAL")
+        return BCType::THERMAL;
+
+    if (s == "CLOSED")
+        return BCType::CLOSED;
+
+    throw std::invalid_argument("Not recognized boundary condition type: " + s);
+}
+
+
+BCComponent component(const std::string& s) {
+    if (s == "OIL")
+        return BCComponent::OIL;
+
+    if (s == "GAS")
+        return BCComponent::GAS;
+
+    if (s == "WATER")
+        return BCComponent::WATER;
+
+    if (s == "SOLVENT")
+        return BCComponent::SOLVENT;
+
+    if (s == "POLYMER")
+        return BCComponent::POLYMER;
+
+    if (s == "NONE")
+        return BCComponent::NONE;
+
+    throw std::invalid_argument("Not recognized boundary condition compononet: " + s);
+}
+
+}
+}
+
+using BCKEY = ParserKeywords::BCPROP;
+BCPROP::BCFace::BCFace(const DeckRecord& record) :
+    index(record.getItem<BCKEY::INDEX>().get<int>(0)),
+    bctype(fromstring::bctype(record.getItem<BCKEY::TYPE>().get<std::string>(0))),
+    component(fromstring::component(record.getItem<BCKEY::COMPONENT>().get<std::string>(0))),
+    rate(record.getItem<BCKEY::RATE>().getSIDouble(0))
+{
+    if (const auto& P = record.getItem<BCKEY::PRESSURE>(); ! P.defaultApplied(0)) {
+        pressure = P.getSIDouble(0);
+    }
+    if (const auto& T = record.getItem<BCKEY::TEMPERATURE>(); ! T.defaultApplied(0)) {
+        temperature = T.getSIDouble(0);
+    }
+}
+
+BCPROP::BCFace BCPROP::BCFace::serializationTestObject()
+{
+    BCFace result;
+    result.index = 100;
+    result.bctype = BCType::RATE;
+    result.component = BCComponent::GAS;
+    result.rate = 101.0;
+    result.pressure = 102.0;
+    result.temperature = 103.0;
+
+    return result;
+}
+
+
+bool BCPROP::BCFace::operator==(const BCPROP::BCFace& other) const {
+    return this->index == other.index &&
+           this->bctype == other.bctype &&
+           this->component == other.component &&
+           this->rate == other.rate &&
+           this->pressure == other.pressure &&
+           this->temperature == other.temperature;
+}
+
+
+
+BCPROP::BCPROP(const Deck& /*deck*/) {
+// TODO Read the indices from BCCON and initial with NONE type???
+}
+
+void BCPROP::updateBCProp(const DeckRecord& record) {
+    const BCPROP::BCFace bcnew( record );
+    for (auto& bc : m_faces) {
+        if (bc.index == bcnew.index)
```

Comment:
This does not allow multiple BCPROP records to set different but compatible conditions. For example a DIRICHLET pressure and THERMAL temperature on the same boundary face. I think the proper semantics here would be to replace it only if index (i.e. location) and type *both* match.

Then we can extend the simulator code later to deal with multiple different conditions on the same face. I believe this will be wanted for the extension to mechanics for example.